istioctl/cmd/istioctl/gendeployment/cmd.go

```diff
@@ -108,9 +104,6 @@ type installation struct {
 	CaTag    string
 	ProxyTag string
```

Comment:
@sdake can you help check if we also need a `kubectlTag`? As the `istioctl gen-deploy --values myvalues.yaml` will generate a YAML, and what is the `kuebctl` version if no tag for it?

Reply:
kubectl tag uses the generalized Istio tag.  We could tag based upon kubectl version, however, this is difficult to know until the container is built.  I will test the `istioctl gen-deploy --values myvalues.yaml` scenario.  Clearly something isnt' quite right about this PR as the gate is failing.

Reply:
looks like istioctl gendeploy is DOA: ```sdake@falkor-07:~/go/src/istio.io/istio/install/kubernetes/helm/istio$ istioctl gen-deploy --values /tmp/values-istio-mcp.yaml 
Error: render error in "istio/templates/crds.yaml": template: istio/templates/crds.yaml:1:72: executing "istio/templates/crds.yaml" at <.Capabilities.Tiller...>: can't evaluate field SemVer in type *version.Version
```

Reply:
after deleting the first and last line of the crd.yaml (the capabilities functionality):
```
      containers:
        - name: kubectl
          image: "gcr.io/istio-release/kubectl:master-latest-daily"
```

gcrio/istio-release/kubectl:master-latest-daily should be generated and pushed by the daily job.  I think this looks good, but the crd.yaml needs attention for gendeploy (or a plan B needs to happen for gendeploy).

Cheers
-steve

Reply:
After inspecting `gen-deploy` `cmd.go', the code looks unmaintained.  For example, there is no galley or sidecarinjector tag and the tags are hidden.  Is anyone actually using `gen-deploy`?  If so, we should probably fix that independently of this PR.

Reply:
Shall we open an issue to talk how to maintain this file? Then I can lgtm this one and got it merged.

I think it maybe ok to delete the subcommand of `gen-deploy`.

/cc @ZackButcher @rshriram

Reply:
Issue as requested: https://github.com/istio/istio/issues/8940